dev/EnvironmentManager/ChangeTracker/EnvironmentVariableChangeTracker.cpp

```diff
@@ -2,37 +2,14 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 #include <pch.h>
 #include "EnvironmentVariableChangeTracker.h"
+#include "EnvironmentVariableChangeTrackerHelper.h"
 
 namespace winrt::Microsoft::Windows::System::implementation
 {
-    EnvironmentVariableChangeTracker::EnvironmentVariableChangeTracker(const std::wstring& key, const std::wstring& valueToSet, EnvironmentManager::Scope scope)
+    EnvironmentVariableChangeTracker::EnvironmentVariableChangeTracker(const std::wstring& evName, const std::wstring& valueToSet, EnvironmentManager::Scope scope)
+        : m_eVName(evName), m_eVValue(valueToSet), m_Scope(scope)
     {
-        THROW_HR_IF(E_INVALIDARG, key.empty());
-
-        // Check if we need to track the changes.
-        // If we do need to track the changes get the Package Full Name
-        WCHAR packageFullName[PACKAGE_FULL_NAME_MAX_LENGTH + 1]{};
-        UINT32 packageFullNameLength{ static_cast<UINT32>(ARRAYSIZE(packageFullName)) };
-        const long getPackageFullNameRC{ ::GetCurrentPackageFullName(&packageFullNameLength, packageFullName) };
-
-
-        if (scope == EnvironmentManager::Scope::Process || getPackageFullNameRC == APPMODEL_ERROR_NO_PACKAGE)
-        {
-            m_ShouldTrackChange = false;
-        }
-        else if (getPackageFullNameRC == ERROR_SUCCESS)
-        {
-            m_PackageFullName = std::wstring(packageFullName);
-            m_ShouldTrackChange = true;
-        }
-        else
-        {
-            THROW_WIN32(getPackageFullNameRC);
-        }
-
-        m_Scope = scope;
-        m_Key = key;
-        m_Value = valueToSet;
+        m_PackageFullName = PackageFullName();
```

Comment:
Why not list in the ctor-initializer-list as `m_PackageFullName()` or simply omit if its default void ctor is good enough for you?

Reply:
I don't understand this question. `m_PackageFullName(PackageFullName())` can't be used in the init-list.